v2/cli/init_command.go

```diff
@@ -0,0 +1,230 @@
+			Name:    flagGit,
+			Aliases: []string{"g"},
+			Usage: "Optionally specify a git repository where the" +
+				" event handling script can be found",
+		},
+	},
+	Action: initialize,
+}
+
+// nolint: gocyclo
+func initialize(c *cli.Context) error {
+
+	// Initialize fields to fill in the template yaml files
+	fields := struct {
+		ProjectID   string
+		Language    string
+		GitCloneURL string
+		Script      string
+	}{
+		c.String(flagID),
+		strings.ToLower(c.String(flagLanguage)),
+		c.String(flagGit),
+		"",
+	}
+
+	err := core.ValidateProjectID(fields.ProjectID)
+	if err != nil {
+		return err
+	}
+
+	// Check if input language is valid
+	fields.Language, err = canonicalizeLanguage(fields.Language)
+	if err != nil {
+		return err
+	}
+
+	// Check if the .brigade directory exists
+	err = file.EnsureDirectory(".brigade")
+	if err != nil {
+		return err
+	}
+
+	// Generate js/ts buffer
+	var scriptBytes []byte
+	scriptBytes, err = execTemplate(scriptTemplate, fields)
+	if err != nil {
+		return err
+	}
+	fields.Script = string(scriptBytes)
+
+	brigadeScriptFileName := fmt.Sprintf("brigade.%s", fields.Language)
+
+	// If git repo is provided, create a separate js/ts file for script
+	var overwrite bool
+	if fields.GitCloneURL != "" {
+		scriptPath := path.Join(".brigade", brigadeScriptFileName)
+		if overwrite, err = checkFileOverwrite(scriptPath); err != nil {
+			return err
+		} else if overwrite {
+			if err = ioutil.WriteFile(
+				scriptPath,
+				scriptBytes,
+				0644,
+			); err != nil {
+				return err
+			}
+		}
+	}
+
+	// Generate the project.yaml file
+	var projectBytes []byte
+	projectBytes, err = execTemplate(projectTemplate, fields)
+	if err != nil {
+		return err
+	}
+
+	// Check if the .brigade/project.yaml path exists.
+	// Confirm overwrite if so.
+	projectPath := path.Join(".brigade", "project.yaml")
+	if overwrite, err = checkFileOverwrite(projectPath); err != nil {
+		return err
+	} else if overwrite {
+		if err = ioutil.WriteFile(
+			projectPath,
+			projectBytes,
+			0644,
+		); err != nil {
+			return err
+		}
+	}
+
+	var notesBytes []byte
+	notesBytes, err = execTemplate(notesTemplate, fields)
+	if err != nil {
+		return err
+	}
+
+	// Create a NOTES.txt file
+	notesPath := path.Join(".brigade", "NOTES.txt")
+	if overwrite, err = checkFileOverwrite(notesPath); err != nil {
+		return err
+	} else if overwrite {
+		if err = ioutil.WriteFile(
+			notesPath,
+			notesBytes,
+			0644,
+		); err != nil {
+			return err
+		}
+	}
+
+	var secretsBytes []byte
+	secretsBytes, err = execTemplate(secretsTemplate, fields)
+	if err != nil {
+		return err
+	}
+
+	// Create a secrets.yaml file
+	secretsPath := path.Join(".brigade", "secrets.yaml")
+	if overwrite, err = checkFileOverwrite(secretsPath); err != nil {
+		return err
+	} else if overwrite {
+		if err = ioutil.WriteFile(
+			secretsPath,
+			secretsBytes,
+			0644,
+		); err != nil {
+			return err
+		}
+	}
+
+	err = addToGitignore(".brigade/secrets.yaml", "./.gitignore")
```

Comment:
There's a second path we should add here. It's `.brigade/node_modules/`

Reply:
I don't actually create a node_modules folder, so are we just assuming the user will install some node dependencies with their project? That should be a safe assumption to make